flink-cdc-common/src/main/java/org/apache/flink/cdc/common/sink/HashFunctionProvider.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.cdc.common.sink;
+
+import org.apache.flink.cdc.common.schema.Schema;
+
+import java.io.Serializable;
+
+/** use for PrePartitionOperator. */
```

Comment:
Provide {@link HashFunction} to help {@link PrePartitionOperator} to shuffle DataChangeEvent to designated subtask. This is usually beneficial for load balancing, when writing to different partitions/buckets in {@link DataSink}, add custom Implementation to further improve efficiency.

Reply:
Sure, thank you for perfect comments so clear.